oshi-core/src/main/java/oshi/software/os/unix/freebsd/FreeBsdOSProcess.java

```diff
@@ -332,25 +332,15 @@ private int queryBitness() {
 
     @Override
     public List<OSThread> getThreadDetails() {
-        List<OSThread> threads = new ArrayList<>();
         String psCommand = "ps -awwxo " + PS_THREAD_COLUMNS + " -H";
         if (getProcessID() >= 0) {
             psCommand += " -p " + getProcessID();
         }
-        List<String> threadList = ExecutingCommand.runNative(psCommand);
-        if (threadList.size() > 1) {
-            // remove header row
-            threadList.remove(0);
-            // Fill list
-            for (String thread : threadList) {
-                Map<PsThreadColumns, String> threadMap = ParseUtil.stringToEnumMap(PsThreadColumns.class, thread.trim(),
-                        ' ');
-                if (threadMap.containsKey(PsThreadColumns.PRI)) {
-                    threads.add(new FreeBsdOSThread(getProcessID(), threadMap));
-                }
-            }
-        }
-        return threads;
+        return ExecutingCommand.runNative(psCommand).stream().skip(1).parallel()
+            .map(thread -> ParseUtil.stringToEnumMap(PsThreadColumns.class, thread.trim(), ' '))
+            .filter(threadMap -> threadMap.containsKey(PsThreadColumns.PRI))
```

Comment:
This one's local to the class, but the purpose here is to make sure we get all the columns so extracting this predicate to a constant would improve readability.